### PR TITLE
Create code-based git diff docs

### PR DESCRIPTION
**Pull request by alexanderahn**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207584391856694

### CODE REVIEW

File: docs/metadata/git/git-commit.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - metadata
+  - Git
+  - commit
+---
+
+This page describes how to commit your changes and create new branches from the main branch.
+
+## Commit changes into the dev branch
+
+When you create your first Project, `dev` branch is automatically created for you. This is to ensure that no edits/commits are made directly on the `main` [branch](/concepts/Project/#Project-commits).
```
Comment (kirankaranth) [inline docs/metadata/git/git-commit.md:16]: This is incorrect. We do not create any branches automatically for the user

Comment (kirankaranth) [reply docs/metadata/git/git-commit.md:16]: We only auto-suggest dev/$username for new branch creation if it is a valid branch

File: docs/metadata/git/git-fork.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - Git
+  - fork
+---
+
+This page describes how to fork a repository in cases where you may want to create a copy of the original repository.
+
+## Use fork per user Git storage model
+
+During the creation of Project there is an option to choose fork per usage as Git Storage model.
+When you fork a repository, you create a copy of the original repository (upstream repository) but the repository remains on your GitHub account.
```
Comment (kirankaranth) [inline docs/metadata/git/git-fork.md:17]: Prophecy does not create any new repository. We instead ask the user to provide both upstream repo and a fork per user repo, both of which have to be already present. The user needs to make sure that the forked repo they input is actually a fork of whatever they input as upstream, as from Prophecy's perspective these are just 2 repos and we cannot verify if one is actually a fork of the other

File: docs/metadata/git/git-resolve.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+<iframe src="https://user-images.Githubusercontent.com/103921419/174733075-b878a77d-97df-4bc9-8baf-5fcc72e268db.mp4" title="How to resolve conflicts" allow="autoplay;fullscreen" allowtransparency="true" frameborder="0" scrolling="no" class="wistia_embed" name="wistia_embed" msallowfullscreen width="100%" height="100%"></iframe>
+</div></div>
+
+### Resolve conflicts using manual merge
+
+For SQL, you can use manual merge to resolve conflicts. This provides you with a simple, yet effective way to resolve merge conflicts for granular changes.
+
+![Choose a Git conflict manual merge](img/git-conflict-manual-merge.png)
+
+- **(A)** **Strategy** - You must choose a preferred strategy to resolve the conflict. Here the Left strategy keeps the version on branch `master`, while the Right strategy keeps the version on branch `dev`.
```
Comment (kirankaranth) [inline docs/metadata/git/git-resolve.md:37]: The left/right strategy is available for resolving conflicts in spark as well. The difference is manual merge by making code changes directly is available only in sql

File: docs/metadata/git/git-resolve.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- **(B)** **Open on master** - Clicking this opens the Model on branch `master` for you to view.
+- **(C)** **Open on dev** - Clicking this opens the Model on branch `dev` for you to view.
+
+Here are the read-only views on branch `master` on the left and branch `dev` on the right:
+
+![View Git conflict merge strategies](img/git-conflict-merge-strategy.png)
+
+You can also toggle on **Code Changes** to view the code before validating. Once you choose the strategy that you want to keep, click **Next**. The merge process will compile the files.
```
Comment (kirankaranth) [inline docs/metadata/git/git-resolve.md:45]: The code changes toggle and the left/right merge strategies are two different separate ways to resolve merge conflicts. Making changes in left/right view does not automatically update the code changes view. Both view are independent and user can choose one of the options to resolve conflicts

File: docs/metadata/git/git-commit.md (hunk 1 above)
Comment (alexanderahn) [reply docs/metadata/git/git-commit.md:16]: Done.

File: docs/metadata/git/git-fork.md (hunk 2 above)
Comment (alexanderahn) [reply docs/metadata/git/git-fork.md:17]: Done.

File: docs/metadata/git/git-resolve.md (hunk 3 above)
Comment (alexanderahn) [reply docs/metadata/git/git-resolve.md:37]: Done.

File: docs/metadata/git/git-resolve.md (hunk 4 above)
Comment (alexanderahn) [reply docs/metadata/git/git-resolve.md:45]: Done.

**Pull request merged**